### PR TITLE
[WIP] SL-920

### PR DESCRIPTION
**Pull request by gary-sedgwick**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Jenkinsfile_parameterized [hunk 1]
```diff
@@ -5,12 +5,52 @@ properties([
...
 
-withParameterizedPipeline(params.TYPE, params.PRODUCT_NAME, params.APP, params.ENVIRONMENT, params.SUBSCRIPTION) {
+product = params.PRODUCT_NAME
+component = params.APP
+environment = params.ENVIRONMENT
+subscription = params.SUBSCRIPTION
+vm_size = params.VM_SIZE
+rules_engine_subnet = params.RULES_ENGINE_SUBNET_NAME
+planOnly = params.PLAN_ONLY
+vaultName = "infra-vault-nonprod"
```
Comment (kpakur) [inline Jenkinsfile_parameterized]: where is that vault coming from?

File: Jenkinsfile_parameterized [hunk 2]
```diff
@@ -5,12 +5,52 @@ properties([
...
+    env.TF_VAR_rules_engine_subnet = "${rules_engine_subnet}"
+    env.TF_VAR_subscription_id = az "keyvault secret show --vault-name '${vaultName}' --name AZURE-SUBSCRIPTION --query value -o tsv"
+    env.TF_VAR_FR_azure_client_id = az "keyvault secret show --vault-name '${vaultName}' --name AZURE-CLIENT-ID --query value -o tsv"
+    env.TF_VAR_FR_azure_client_secret = az "keyvault secret show --vault-name '${vaultName}' --name AZURE-CLIENT-SECRET --query value -o tsv"
+    env.TF_VAR_consul = az "vmss nic list --resource-group 'core-infra-${environment}' --vmss-name consul-server --query '[0].ipConfigurations[0].privateIpAddress'"
+    env.TF_VAR_FR_vaultname = "${vaultName}"
+
+    stage("spinInfra") {
+      env.TF_VAR_netnum = findFreeSubnet(params.SUBSCRIPTION, params.ENVIRONMENT)[1]
+      sh "pwd"
```
Comment (kpakur) [inline Jenkinsfile_parameterized]: ?

File: Jenkinsfile_parameterized [hunk 3]
```diff
@@ -5,12 +5,52 @@ properties([
...
+
+echo "Building '${product}-${component}' in '${environment}'"
+
+node {
+  env.PATH = "$env.PATH:/usr/local/bin"
+  def az = { cmd -> return sh(script: "env AZURE_CONFIG_DIR=/opt/jenkins/.azure-$subscription az $cmd", returnStdout: true).trim() }
+
+
+  stage("checkout") {
+    stageCheckout('git@github.com:hmcts/snl-rules.git')
```
Comment (kpakur) [inline Jenkinsfile_parameterized]: this is problematic, why just not simply "checkout" (without the url) then it should check it out from what jenkins is running it for, see also https://jenkins.io/doc/book/pipeline/jenkinsfile/

File: Jenkinsfile_parameterized (hunk 1 above)
Comment (gary-sedgwick) [reply Jenkinsfile_parameterized]: So this file was pretty much a copy of what Dan had done on the original iaas branch. That vault exists, but I haven't drilled down into the purpose or what it's there for - I had assumed it was necessary for the VM config. I'll look into the details and get back to you, same for other changes in this file.

**Pull request closed without merging**